vscode-ext/src/extension.ts

```diff
@@ -1,17 +1,41 @@
 import { CoreV1Api, V1NamespaceList, V1PodList } from '@kubernetes/client-node';
-import { settings } from 'cluster';
 import * as vscode from 'vscode';
-
+import YAML from 'yaml';
+const TOML = require('toml');
 const semver = require('semver');
 const https = require('https');
 const k8s = require('@kubernetes/client-node');
 const path = require('path');
 const os = require('os');
+const glob = require('glob');
 
 const LIBRARIES: { [platform: string]: [var_name: string, lib_name: string] } = {
 	'darwin': ['DYLD_INSERT_LIBRARIES', 'libmirrord_layer.dylib'],
 	'linux': ['LD_PRELOAD', 'libmirrord_layer.so']
 };
+
+const DEFAULT_CONFIG = `
+{
+    "accept_invalid_certificates": false,
+    "feature": {
+        "network": {
+            "incoming": "mirror",
+            "outgoing": true
+        },
+        "fs": "write",
+        "env": true
+    }
+}
+`;
+
+const VS_CODE_DIR = function() {
+	let folders = vscode.workspace.workspaceFolders;
+	if (folders === undefined) {
+		throw new Error('No workspace folder found');
+	}
```

Comment:
instead of throwing an exception, we should catch and spawn a notification

Reply:
And then what? It's non-recoverable

Reply:
even though its not, we can let the user know what happened

Reply:
VS Code does that for you when you throw an exception